### DISCUSSION TITLE
Repo language issue

### DISCUSSION THREAD

**Discussion by souvikde99** [community/community#23755] [New to GitHub]
<p>i have uploaded an html file, but github repo is taking it as java. please help. my website is not displaying properly</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there! :wave: Welcome to the Community!</p>
<p>When you say “the github repo is taking it as java” I think you’re talking about the language bar that displays on the front of the repository? GitHub determines language statistics based on the total size of the files that belong to each language.</p>
<p>We use the Linguist open source library to analyze source code files. You can read about how to use overrides in the Linguist documentation:</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/github/linguist#troubleshooting" target="_blank">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/19803303/181079236-c9992b1c-392b-4d38-9030-1acff912464d.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/github/linguist#troubleshooting" target="_blank">github/linguist</a></h3>

<p>Language Savant. If your repository's language is being reported incorrectly, send us a pull request! - github/linguist</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>…and create a <code>.gitattributes</code> file that specifies your file should be HTML, not Java.</p>
<p>That won’t affect how your website displays, assuming you’re talking about a GitHub Pages site. If the repo is public can you give me a link to it and I’ll take a look!</p>
